### PR TITLE
[BWC and API enforcement] Decorate the existing APIs with proper annotations (part 2)

### PR DESCRIPTION
**Pull request by reta**
### Description
 Decorate the existing APIs with proper annotations. This is merely the first part so we could work through the small changesets vs big bang pull request. The API validations are fully automated (as annotation processor https://github.com/opensearch-project/OpenSearch/issues/9304). 

All classes that are annotated in this pull request leak through `org.opensearch.client.Client` which is by definition (currently) is public APIs exposed through plugins and RestClient / HighLevelRestClient. 

### Related Issues
Part of https://github.com/opensearch-project/OpenSearch/issues/9303

### Check List
- [ ] ~New functionality includes testing.~
  - [ ] ~All tests pass~
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [ ] ~Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@andrross @nknize @dblock another portion of annotated APIs - I have checked all of them and each leaks through the roots that have public API annotation. I am thinking about one more pull request before adding the annotation processor to the build (to catch leaking APIs automatically), it is ready but if we opt in to it now, it will fail the build. Thanks folks!

**Comment by reta** [comment]
> * 
> 
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/29445/

https://github.com/opensearch-project/OpenSearch/issues/7934
https://github.com/opensearch-project/OpenSearch/issues/8771

File: server/src/main/java/org/opensearch/index/query/Rewriteable.java [hunk 1]
```diff
@@ -41,8 +42,9 @@
 /**
  * A basic interface for rewriteable classes.
  *
- * @opensearch.internal
+ * @opensearch.api
  */
+@PublicApi(since = "1.0.0")
 public interface Rewriteable<T> {
```
Comment (reta) [inline server/src/main/java/org/opensearch/index/query/Rewriteable.java:48]: May look suspicious ... but leaks from `ShardSearchRequest` ....

File: server/src/main/java/org/opensearch/search/internal/AliasFilter.java [hunk 2]
```diff
@@ -47,8 +48,9 @@
 /**
  * Represents a {@link QueryBuilder} and a list of alias names that filters the builder is composed of.
  *
- * @opensearch.internal
+ * @opensearch.api
  */
+@PublicApi(since = "1.0.0")
 public final class AliasFilter implements Writeable, Rewriteable<AliasFilter> {
```
Comment (reta) [inline server/src/main/java/org/opensearch/search/internal/AliasFilter.java:54]: Leaks through `ExplainRequest` :(

**Comment by reta** [comment]
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/29477/

https://github.com/opensearch-project/OpenSearch/issues/10006

**Comment by reta** [comment]
@andrross @nknize @dblock friendly ping folks so we could move forward, thank you

**Review by andrross — APPROVED** [review]

**Pull request merged**